Makefile

```diff
@@ -1,15 +1,47 @@
+VE ?= ./ve
+FLAKE8 ?= $(VE)/bin/flake8
+REQUIREMENTS ?= requirements.txt
+SYS_PYTHON ?= python3
+ENV_PYTHON ?= $(VE)/bin/python
+PIP ?= $(VE)/bin/pip
+PY_SENTINAL ?= $(VE)/sentinal
+WHEEL_VERSION ?= 0.29.0
+VIRTUALENV ?= virtualenv.py
+SUPPORT_DIR ?= requirements/virtualenv_support/
+SCRIPT_FILE ?= convert.py
+MAX_COMPLEXITY ?= 10
+PY_DIRS ?= *.py tests --exclude virtualenv.py
+
 #STAGING_URL=https://histologylab-static.stage.ctl.columbia.edu/
 #PROD_URL=https://histologylab.ctl.columbia.edu/
 #STAGING_BUCKET=histologylab-static.stage.ctl.columbia.edu
 #PROD_BUCKET=histologylab.ctl.columbia.edu
-INTERMEDIATE_STEPS ?= make $(PUBLIC)/js/all.json
+#INTERMEDIATE_STEPS ?= make $(PUBLIC)/js/all.json
 
 JS_FILES=themes/ctl-histologylab/static/js/
 
 all: eslint
 
 include *.mk
 
-$(PUBLIC)/js/all.json: $(PUBLIC)/json/all/index.html
-	mv $< $@ \
-	&& ./checkjson.py
+$(PY_SENTINAL): $(REQUIREMENTS) $(VIRTUALENV) $(SUPPORT_DIR)* 
+	rm -rf $(VE)
+	$(SYS_PYTHON) $(VIRTUALENV) --extra-search-dir=$(SUPPORT_DIR) --never-download $(VE)
```

Comment:
If you're only using python 3, we can simplify this, and remove `virtualenv.py`. See the note on virtualenv here: https://github.com/ccnmtl/django-lti-provider/pull/88

You should have a `pyvenv` command available if your system has python3, and you can just replace `$(SYS_PYTHON) $(VIRTUALENV)` here with `pyvenv`.

Reply:
If this is confusing though, we can definitely just use the existing `virtualenv.py` setup here.

Reply:
There's an example of this in blackrock_fetcher where I've converted over to the python3 style virtualenv: https://github.com/ccnmtl/blackrock_fetcher/pull/53/files

`pyvenv` is actually deprecated, and the current recommended way to make a virtualenv is with `python3 -m venv ve`.

Reply:
Cool, I didn't know that. I'll swap it out.

Reply:
If you see any problems with what I suggested, just leave it as is though. I think there are still some things to work out. I'll document this once I understand it more. I tried doing this conversion on `econplayground` and it looks like there's a problem with pip not being available: https://travis-ci.org/ccnmtl/econplayground/builds/314062734#L812